src/main/java/HW22DB/LessonDao.java

```diff
@@ -0,0 +1,123 @@
+
+            int rows = statement.executeUpdate();
+
+            if (rows == 0) {
+                throw new CustomerCreateException("Unable to create customer");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                lesson.setId(generatedKeys.getInt(1));
+            }
+
+        } catch (SQLException | CustomerCreateException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void delLesson(String name1) {
+        try {
+            PreparedStatement statement = connection.prepareStatement("DELETE FROM lesson WHERE name1=?");
+            statement.setString(1, name1);
+            statement.execute();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public List<Lesson> showAllLessons() {
+        Statement statement = null;
+        List<Lesson> lessons = new ArrayList<>();
+        try {
+            statement = connection.createStatement();
+            boolean dateReturned = statement.execute("SELECT * FROM lesson");
+            if (dateReturned) {
+                ResultSet resultSet = statement.getResultSet();
+                while (resultSet.next()) {
+                    Lesson lesson = new Lesson();
+                    lesson.setId(resultSet.getInt("id"));
+                    lesson.setName1(resultSet.getString("name1"));
+                    lesson.setUpdatedAt(resultSet.getDate("updateAT"));
+                    lesson.setHomework_id(resultSet.getInt("homework_id"));
+                    lessons.add(lesson);
+                }
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+        return lessons;
+    }
+
+    public List<Object> showLessonByID(Integer id) {
+        try {
+            PreparedStatement statement = connection.prepareStatement("select * from lesson Left join homework on lesson.id= homework.id where lesson.id=?");
+            statement.setInt(1, id);
+            boolean dateReturned = statement.execute();
+            if (dateReturned) {
+                ResultSet resultSet = statement.getResultSet();
+                List<Object> objects = new ArrayList<>();
+                while (resultSet.next()) {
+                    Lesson lesson = new Lesson();
+                    lesson.setId(resultSet.getInt("id"));
+                    lesson.setName1(resultSet.getString("name1"));
+                    lesson.setUpdatedAt(resultSet.getDate("updateAT"));
+                    lesson.setHomework_id(resultSet.getInt("homework_id"));
+                    Homework homework = new Homework();
+                    homework.setId(resultSet.getInt("id"));
+                    homework.setName1(resultSet.getString("name1"));
+                    homework.setDescription(resultSet.getString("description"));
+                    objects.add(lesson);
```

Comment:
instead of this use one-to-one relationship.
This code can be replaced with `lesson.setHomework(homework)`